### PR TITLE
check currentNavigate.externalUrl for url matching to address #349

### PR DESCRIPTION
**Pull request by lingyan**
@skippykawakami

cc: @mridgway @redonkulus 

An alternative to #351 for fixing a Safari edge case reported in #349.  Instead of messing with detecting popstate event on page load for safari browser only (which requires user agent parsing that I want to avoid), or messing with browser history, this approach is safer.

##### Work needed on app side:
* dispatch `externalUrl` as part of the payload for the NAVIGATE_START event on server side.  For client side navigation, `externalUrl` field is not needed and will not be set.

##### Notes:
* I would still strongly suggest you to get your ATS team to change to pass HTTP header instead of query param, if they can :)
* I haven't updated doc for this, because I'd like to keep this as an undocumented feature for now.

Give it a try and let me know what you think.  Thanks!

### CODE REVIEW

**Comment by yahoocla** [comment]
Thank you for submitting this pull request, however I do not see a valid CLA on file for you. Before we can merge this request please visit https://yahoocla.herokuapp.com/ and agree to the terms. Thanks! :smile:

**Comment by yahoocla** [comment]
CLA is valid!

File: packages/fluxible-router/lib/handleHistory.js [hunk 1]
```diff
@@ -88,7 +88,7 @@ function createComponent(Component, opts) {
                 //   for the route represented by the hash fragment.
 
                 var urlFromHistory = this._history.getUrl();
-                var urlFromState = this.props.currentRoute && this.props.currentRoute.get('url');
+                var urlFromState = this.props.currentRoute && this.props.currentRoute.url;
```
Comment (lingyan) [inline packages/fluxible-router/lib/handleHistory.js]: this is a bug fix not directly related to #349

File: packages/fluxible-router/lib/handleHistory.js [hunk 2]
```diff
@@ -121,14 +121,21 @@ function createComponent(Component, opts) {
...
-            var currentRouteMap = props.currentRoute || {};
-            var currentUrl = currentRouteMap.url;
+            var currentRoute = props.currentRoute || {};
+            var nav = props.currentNavigate || {};
+
+            // Add currentNavigate.clientRelativeUrl for https://github.com/yahoo/fluxible/issues/349:
+            // "Safari popstate issue causing handleHistory.js to execute the navigateAction on page load".
+            // This needs app to dispatch "clientRelativeUrl" as part of the payload for the NAVIGATE_START event
+            // on server side.  For client side navigation, this field is not needed and will not be set.
+            var currentUrl = nav.clientRelativeUrl || currentRoute.url;
```
Comment (lingyan) [inline packages/fluxible-router/lib/handleHistory.js]: @skippykawakami For example, if request url in browser is http://mydomain.com/abc, and ATS changes it to http://mydomain.com/abc?a=1 for whatever reason, (**NOTE** using query param is not recommended.  should use HTTP header instead.), your app can dispatch `NAVIGATE_START` event with payload like this:
```
{
   url: '/abc',
  method: 'GET',
  clientRelativeUrl: '/abc?a=1'
}
```

File: packages/fluxible-router/lib/handleHistory.js (hunk 1 above)
Comment (mridgway) [inline packages/fluxible-router/lib/handleHistory.js]: We should do this in a separate PR.

Comment (mridgway) [inline packages/fluxible-router/lib/handleHistory.js]: Nvm, looks like it was already done.

**Comment by kaesonho** [comment]
:+1:
